lib/src/utf8_backed_string.dart

```diff
@@ -3,21 +3,23 @@ import 'dart:convert';
 import 'package:buffer/buffer.dart';
 
 class UTF8BackedString {
```

Comment:
I think the changes to the implementation look reasonable, but `UTF8BackedString` is no longer a suitable class name for what this is doing.

Maybe rename it to `EncodedString` and rename the getters to something like `byteLength` and `encodedBytes`?